VerbalExpressions/VerbalExpressions.swift

```diff
@@ -21,84 +21,132 @@ public class VerbalExpressions {
 
     // computed properties
     var pattern: String { return prefixes + source + suffixes }
-    
+
     var regularExpression: NSRegularExpression! {
-        get {
-            let regex = try! NSRegularExpression(pattern: pattern, options: options)
-            
-            return regex
-        }
+        return try! NSRegularExpression(pattern: pattern, options: options)
     }
 
     // instance methods
-    public func startOfLine(enabled enabled: Bool = true) -> Self {
-        prefixes = enabled ? "^" : ""
-
-        return self
+    public func startOfLine(enabled enabled: Bool = true) -> VerbalExpressions {
+        return VerbalExpressions(
+            prefixes: enabled ? "^" : "",
+            source: self.source,
+            suffixes: self.suffixes,
+            options: self.options
+        )
     }
 
-    public func endOfLine(enabled enabled: Bool = true) -> Self {
-        suffixes = enabled ? "$" : ""
-
-        return self
+    public func endOfLine(enabled enabled: Bool = true) -> VerbalExpressions {
+        return VerbalExpressions(
```

Comment:
Personally I think Lenses may make the code harder to read more many people, but it would reduce boilerplate.
I'll give it a try.